exporters/zipkin/include/opentelemetry/exporters/zipkin/recordable.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include "nlohmann/json.hpp"
+#include "opentelemetry/sdk/trace/recordable.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace zipkin
+{
+using ZipkinSpan = nlohmann::json;
+
+enum class TransportFormat
+{
+  JSON,
+  PROTOBUF
+};
+
+class Recordable final : public sdk::trace::Recordable
+{
+public:
+  const ZipkinSpan &span() const noexcept { return span_; }
+
+  void SetIds(trace::TraceId trace_id,
+              trace::SpanId span_id,
+              trace::SpanId parent_span_id) noexcept override;
+
+  void SetAttribute(nostd::string_view key,
+                    const opentelemetry::common::AttributeValue &value) noexcept override;
+
+  void AddEvent(nostd::string_view name,
+                core::SystemTimestamp timestamp,
+                const common::KeyValueIterable &attributes) noexcept override;
+
+  void AddLink(const opentelemetry::trace::SpanContext &span_context,
+               const common::KeyValueIterable &attributes) noexcept override;
+
+  void SetStatus(trace::CanonicalCode code, nostd::string_view description) noexcept override;
```

Comment:
Update `CanonicalCode` to `StatusCode`?

Reply:
Yes, plan to do it based on which PR is merged first :)